lib/matplotlib/cm.py

```diff
@@ -177,23 +136,67 @@ def register(self, cmap, *, name=None, force=False):
             registered name. True supports overwriting registered colormaps
             other than the builtin colormaps.
         """
+        if not isinstance(cmap, colors.Colormap):
+            raise ValueError("You must pass a Colormap instance. "
+                             f"You passed {cmap} a {type(cmap)} object.")
+
         name = name or cmap.name
-        if name in self and not force:
-            raise ValueError(
-                f'A colormap named "{name}" is already registered.')
-        register_cmap(name, cmap.copy())
+        if name in self:
+            if not force and name in self._builtin_cmaps:
+                msg = f"Trying to re-register the builtin cmap {name!r}."
+                raise ValueError(msg)
+            elif not force:
+                raise ValueError(
+                    f'A colormap named "{name}" is already registered.')
+            else:
+                _api.warn_external(f"Trying to register the cmap {name!r} "
+                                   "which already exists.")
 
+        self._cmaps[name] = cmap.copy()
+
+    def unregister(self, name):
+        """
+        Remove a colormap from the registry.
+
+        You may not remove built-in colormaps.
```

Comment:
```suggestion
        You cannot remove built-in colormaps.
```
Not a native speaker: Is that more clear?